src/ethereum/ethEngine.ts

```diff
@@ -751,7 +756,28 @@ export class EthereumEngine
           this.networkInfo
         )
         const fee = mul(gasPrice, gasLimit)
-        const totalAmount = add(mid, fee)
+        let l1Fee = '0'
+
+        if (this.l1RollupParams != null) {
+          const txData: CalcL1RollupFeeParams = {
+            nonce: this.otherData.unconfirmedNextNonce,
+            gasPriceL1Wei: ceil(
+              mul(
+                this.l1RollupParams.gasPriceL1Wei,
+                this.l1RollupParams.maxGasPriceL1Multiplier
+              ),
+              0
```

Comment:
I just saw this same code in the previous commit. Can it be common-ized? If not, no worries, but it needs the same fix as the other place.